From 342c2fdaf94f195f6273a88f9fe430c189e7168f Mon Sep 17 00:00:00 2001
From: Topi Miettinen <topimiettinen@users.noreply.github.com>
Date: Thu, 31 Aug 2017 08:37:32 +0000
Subject: [PATCH] log: reopen log for failed assertions (#6703)

Reopen log so that failed and aborting assertions can be written to log.

Closes: #6658
(cherry picked from commit ea89a119cda917a17bd186b3c13197acfd655b12)
---
 src/basic/log.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/basic/log.c b/src/basic/log.c
index 3fd53800a0..ad99773cbb 100644
--- a/src/basic/log.c
+++ b/src/basic/log.c
@@ -804,6 +804,7 @@ noreturn void log_assert_failed_realm(
                 const char *file,
                 int line,
                 const char *func) {
+        log_open();
         log_assert(LOG_REALM_PLUS_LEVEL(realm, LOG_CRIT), text, file, line, func,
                    "Assertion '%s' failed at %s:%u, function %s(). Aborting.");
         abort();
@@ -815,6 +816,7 @@ noreturn void log_assert_failed_unreachable_realm(
                 const char *file,
                 int line,
                 const char *func) {
+        log_open();
         log_assert(LOG_REALM_PLUS_LEVEL(realm, LOG_CRIT), text, file, line, func,
                    "Code should not be reached '%s' at %s:%u, function %s(). Aborting.");
         abort();
